week4/week4Pllayground.playground/Contents.swift

```diff
@@ -0,0 +1,241 @@
+  func calculateAverageGrade(recentScore: Int) -> Double {
+    var sum = recentScore + grades.reduce(0, +)
+    var average = sum/grades.count
+    return Double(average)
+  }
+  
+  func calcuculateAverageGrade( recentScore: Int, bonusPoints: Int) -> Double {
+    var sum = bonusPoints + recentScore + grades.reduce(0, +)
+    var average = sum/grades.count
+    return Double(average)
+  }
+  
+  
+  func details() {
+    print("Name: \(person.firstName) \(person.lastName), Age: \(person.age), GPA: \(calculateAverageGrade()) ")
+  }
+}
+
+var student = Student(person: Person(firstName: "Nick", lastName: "Ryan", age: 37), grades: [94, 99, 81, 100, 70])
+/* - --------------------------- emd of previous code*/
+
+var updatedGrade = student.calculateAverageGrade(recentScore: Int(89.99))
+student.calculateAverageGrade()
+student.calcuculateAverageGrade(recentScore: 87, bonusPoints: 10)
+
+
+//b) Create an enum called Animal that has at least five animals. Next, make a function called theSoundMadeBy that has a parameter of type Animal. This function should output the sound that the animal makes. For example, if the Animal pass is a cow, the function should output, “A cow goes moooo.” Hint: Do not use if statements to complete this section.
+//Call the function twice, sending a different Animal each time.
+
+
+enum Animals {
+  case kobe
+  case chronic
+  case grey
+  case blanketMonsater
+  case filamentDog
+}
+
+func theSoundMadeBy (animal: Animals) {
+  switch animal {
+  case .kobe:
+    print("Kobe goes woo woo woo")
+  case .chronic:
+    print("Chronic goes grrr......")
+  case .grey:
+    print("Grey goes gr... gr... gr..")
+  case .blanketMonsater:
+    print("Blanket Monster is gonna get you")
+  case .filamentDog:
+    print("Fliament dog goes bzzz.......bzz...")
+  }
+}
+
+theSoundMadeBy(animal: .kobe)
+theSoundMadeBy(animal: .chronic)
+
+//This question will have you creating multiple functions that will require you to use closures and collections. First, you will do some setup.
+//Create an array of Int called nums with the values of 0 to 100.
+//Create an array of Int? called numsWithNil with the following values:
+//79, nil, 80, nil, 90, nil, 100, 72
+//Create an array of Int called numsBy2 with values starting at 2 through 100, by 2.
+//Create an array of Int called numsBy4 with values starting at 2 through 100, by 4.
+
+var nums = Array(1...100)
+var numsWithNil: [Int?] = [79, nil, 80, nil, 90, nil, 100, 72]
+
+var numsBy2: [Int] = []
+for num in stride(from: 2, through: 100, by: 2) {
+  numsBy2.append(num)
+}
+
+var numsBy4: [Int] = []
+for num in stride (from: 2, through: 100, by: 4) {
+  numsBy4.append(num)
+}
+
+//- Create a function called evenNumbersArray that takes a parameter of [Int] (array of Int) and returns [Int]. The array of Int returned should contain all the even numbers in the array passed. Call the function passing the nums array and print the output.
+func evenNumbersArray (arrayOfInt:[Int]) -> [Int] {
```

Comment:
The "(" from the parameter list of the function should not have an extra space between the name of the parameter and it. As a best practice of Swift.